llama_index/node_parser/code_hierarchy.py

```diff
@@ -0,0 +1,284 @@
+
+from typing import Any, List, Optional, Sequence
+
+from pydantic import Field
+from tree_sitter import Node
+from llama_index.callbacks.base import CallbackManager
+from llama_index.callbacks.schema import CBEventType, EventPayload
+from llama_index.node_parser.extractors.metadata_extractors import MetadataExtractor
+from llama_index.node_parser.interface import NodeParser
+from llama_index.node_parser.simple import SimpleNodeParser
+from llama_index.schema import BaseNode, Document, NodeRelationship, TextNode
+from llama_index.text_splitter.code_splitter import CodeSplitter
+from llama_index.utils import get_tqdm_iterable
+from pydantic import BaseModel
+
+# TODO: Expand these for new languages
```

Comment:
Is it possible to put a quick snippet on how these were found/created? Will make adding new languages a little easier in future contributions :)

Reply:
That's a 100% MUST IMO as well.

Unfortunately it was rather difficult. A lot of debugger and breakpoint usage. I was hoping a trend would appear but it appears to be pretty language specific